Source/Utils/WarpXMovingWindow.cpp

```diff
@@ -56,25 +56,82 @@ void
 WarpX::UpdatePlasmaInjectionPosition (amrex::Real a_dt)
 {
     const int dir = moving_window_dir;
-    // Continuously inject plasma in new cells (by default only on level 0)
-    if (WarpX::warpx_do_continuous_injection and (WarpX::gamma_boost > 1)){
-        // In boosted-frame simulations, the plasma has moved since the last
-        // call to this function, and injection position needs to be updated
-        current_injection_position -= WarpX::beta_boost *
+
+    // Loop over species
+    const int n_species = mypc->nSpecies();
+    for (int i=0; i<n_species; i++)
+    {
+        const WarpXParticleContainer& pc = mypc->GetParticleContainer(i);
+        const std::vector<std::string>& species_names = mypc->GetSpeciesNames();
+
+        // Continuously inject plasma in new cells (by default only on level 0)
+        if (pc.doContinuousInjection())
+        {
+            const std::unique_ptr<PlasmaInjector> plasma_injector
```

Comment:
Could we get the member variable `plasma_injector` of the instance of `PhysicalParticleContainer` instead of creating a new one?

Reply:
Similarly to the comment below, `plasma_injector` is a member variable of the class `PhysicalParticleContainer`, but the object from which we would try to get it is `pc` defined as `WarpXParticleContainer& pc = mypc->GetParticleContainer(i);`, unless there is a way to get only the physical particle containers from `mypc`.

Reply:
Discussed offline: let's define a virtual method in `WarpXParticleContainer` to get the plasma injector whenever available in a derived class.

Reply:
Done in [74e333f](https://github.com/ECP-WarpX/WarpX/pull/3958/commits/74e333fb43efa73036798ff9c9aa5dc971a2fa1a).